src/java/com/sarality/app/view/dialog/DialogActionPerformer.java

```diff
@@ -0,0 +1,25 @@
+package com.sarality.app.view.dialog;
+
+import android.content.DialogInterface;
+
+import com.sarality.app.view.action.Action;
+
+/**
+ * Utility class to perform the action.
+ * <p/>
+ * This is the class that gets registered with the dialog and delegates the call to the actual action.
+ *
+ * @author abhideep@ (Abhideep Singh)
+ */
+class DialogActionPerformer implements DialogInterface.OnCancelListener {
```

Comment:
is this eventually going to handle all the listeners to the dialog and handle all the  actions on the dialog?